crates/biome_service/src/workspace/server.rs

```diff
@@ -935,16 +934,13 @@ impl Workspace for WorkspaceServer {
             });
         }
 
-        if params.watch {
-            self.watched_folders.pin().insert(path.clone());
-
-            let _ = self.watcher_tx.try_send(WatcherInstruction::WatchFolder(
-                path.clone(),
-                scan_kind.clone(),
-            ));
-        }
+        let scan_options = ScanOptions {
+            scan_kind,
+            verbose,
+            watch,
+        };
```

Comment:
We move the watcher instructions into the scanner as much as we can. The scanner knows which folders should be scanned, and thus, which ones should be watched.